tests/end_to_end/requirements.txt

```diff
@@ -26,4 +26,4 @@ rich==13.5.2
 six==1.16.0
 SQLAlchemy
 typeguard==2.13.3
-typing_extensions==4.7.1
+typing_extensions==4.10.0
```

Comment:
this version includes the missing function `TypeIs`

Reply:
<img width="631" alt="image" src="https://github.com/user-attachments/assets/274de87d-165f-44be-94c9-0954e113f3a4">